docs/styleguides/python.md

```diff
@@ -0,0 +1,259 @@
+            print("pre-reqs satisfied")
+
+        installer.setup_remote(
+            tunnel=bootstrap_host_tunnel,
+            installer_path=cluster.ssher.home_dir + '/dcos_generate_config.sh',
+            download_url=installer_url)
+
+        if installer.setup.ok:
+            print("remote setup successful.")
+        else:
+            print("remote setup successful.")
+    ```
+
+   * double indented - Recommended # GOOD. Note the visual clarity.
+
+    ```
+    def method():
+        if installer.prereqs.ok:
+            print("pre-reqs satisfied")
+
+        installer.setup_remote(
+                tunnel=bootstrap_host_tunnel,
+                installer_path=cluster.ssher.home_dir + '/dcos_generate_config.sh',
+                download_url=installer_url)
+
+        if installer.setup.ok:
+            print("remote setup successful.")
+        else:
+            print("remote setup successful.")
+    ```
+
+
+* Adding Types. Since our code-base is Python3.5+ compatible, we have the choice of introducing types, validating
+  with mypy.
+
+```
+        class VpcClusterUpgradeTestDcosApiSessionFactory:
+
+            def apply(
+                    self,
+                    dcos_url: str,
+                    masters: List[str],
+                    public_masters: List[str],
+                    slaves: List[str],
+                    public_slaves: List[str],
+                    default_os_user: str) -> DcosApiSession:
+```
+
+* Reviewers will request the author to write types for the method parameters.
+
+    _Rationale_: We will start running mypy type checker as part of build process.
+
+
+* Shebang Rule.
+    * We will consistently use the shebang line in our scripts.
+        * `#!/usr/bin/env python3`
+        OR
+        * `#!/usr/bin/env python2` in our scripts.
+
+    _Rationale_ : Across platforms, and different python versions in a given platform, this is the most consistent way
+    to indicate the interpreter required for your python program.
+
+* Use the `format` method
+    * Prefer `'{named}'.format` will be over `{}.format`.
+        * Rationale: If the number of arguments are long. `{named}.format` helps in readability.
+    * Use your best judgement to decide between + and % (or format) when applicable.
+    * This style guide does not recommend [f-strings](https://www.python.org/dev/peps/pep-0498/) as `f-strings` are
+      still very early feature in Python, yet to be seen wide adoption, and it is available only from
+      Python 3.6 onwards.
+
+     _Rationale_: When there are multiple format specifiers used in a string, `{name1} {name2}` formats are easier
+    to read than `{}{}` (empty) or `{0}{1}` (indexed) formats.
+
+
+
+* TODO Comment Format
+    * Follow the rules on [when to write TODO comments](https://google.github.io/styleguide/pyguide.html?showone=TODO_Comments#TODO_Comments)
+    * Format required will be `TODO (active-developer) [JIRA] - Brief one line description`
```

Comment:
Sounds good to me.  I think, it is changing `-` to `:`

I will use the example that you specified.

```
# TODO(active-developer): DCOS_OSS-1234 - Brief one line description.
```